diff --git a/gpui.info b/gpui.info
index 1daacbc..bca9b0a 100644
--- a/gpui.info
+++ b/gpui.info
@@ -1,7 +1,7 @@
 name = Grammar Parser UI
 description = User interface to the Grammar Parser library.
 package = Development
-core = 7.x
+core = 6.x
 
-dependencies[] = gplib (<2.0)
+dependencies[] = gplib
 
diff --git a/gpui.install b/gpui.install
index b4d3937..5dfe9ff 100644
--- a/gpui.install
+++ b/gpui.install
@@ -15,7 +15,7 @@ module_load_include('inc', 'gpui', 'gpui');
 function gpui_install() {
   // Create the parser directory.
   $dir = gpui_directory_path('parser', FALSE);
-  if (!file_prepare_directory($dir, FILE_CREATE_DIRECTORY)) {
+  if (!file_check_directory($dir, FILE_CREATE_DIRECTORY)) {
     drupal_set_message(st('Your files directory at %directory can not be written to. Grammar Parser UI places parsed source code files in this directory.', array('%directory' => $dir)), 'error');
   }
 }
@@ -48,7 +48,7 @@ function gpui_requirements($phase) {
   // Test writeability to files directory.
   if ($phase == 'install') {
     $dir = gpui_directory_path('', FALSE);
-    if (!file_prepare_directory($dir, FILE_CREATE_DIRECTORY)) {
+    if (!file_check_directory($dir, FILE_CREATE_DIRECTORY)) {
       $requirements['gpui_files'] = array(
         'title' => $t('Files directory'),
         'description' => $t('Your files directory at %directory can not be written to. Grammar Parser places parsed source code files in this directory.', array('%directory' => $dir)),
@@ -58,7 +58,7 @@ function gpui_requirements($phase) {
   }
   else {
     $dir = gpui_directory_path('parser', FALSE);
-    if (!file_prepare_directory($dir, FILE_CREATE_DIRECTORY)) {
+    if (!file_check_directory($dir, FILE_CREATE_DIRECTORY)) {
       $requirements['gpui_files'] = array(
         'title' => $t('Grammar Parser directory'),
         'description' => $t('Your files directory at %directory can not be written to. Grammar Parser places parsed source code files in this directory.', array('%directory' => $dir)),
diff --git a/gpui.module b/gpui.module
index abdb0fe..be9e7b2 100644
--- a/gpui.module
+++ b/gpui.module
@@ -17,14 +17,14 @@
 module_load_include('inc', 'gpui', 'gpui');
 
 /**
- * Implements hook_permission().
+ * Implements hook_perm().
  */
-function gpui_permission() {
-  return array(
-    'manage parser' => array(
+function gpui_perm() {
+  return array( // Revert this to 6.x format
+    'manage parser' /* => array(
       'title' => t('Manage parser'),
       'description' => t('Manage source code parser execution'),
-    ),
+    ),*/
   );
 }
 
@@ -98,7 +98,7 @@ function gpui_menu() {
 function gpui_theme() {
   return array(
     'gpui_outputs_table' => array(
-      'render element' => 'form',
+      'arguments' => array('form' => NULL) // Revert this to 6.x format
     ),
   );
 }
@@ -106,7 +106,7 @@ function gpui_theme() {
 /**
  * Form builder for settings form.
  */
-function gpui_settings_form($form, &$form_state) {
+function gpui_settings_form($form_state) {
   $path = gpui_directory_path('', FALSE);
   $form['gpui_dir'] = array(
     '#type' => 'textfield',
@@ -161,7 +161,7 @@ function gpui_settings_form_submit($form, &$form_state) {
 /**
  * Form builder for the source code parser form.
  */
-function gpui_parser_form($form, &$form_state) {
+function gpui_parser_form($form_state) {
   gpui_check_settings($form_state);
 
   // Set default values.
@@ -355,8 +355,7 @@ function gpui_parser_form_defaults($form_state) {
  *
  * @return HTML output.
  */
-function theme_gpui_outputs_table($variables) {
-  $form = $variables['form'];
+function theme_gpui_outputs_table($form) {
   if (function_exists('theme_table_select_header_cell')) {
     // For D6.
     $select_header = theme('table_select_header_cell');
@@ -379,13 +378,13 @@ function theme_gpui_outputs_table($variables) {
     $rows[] = $row;
   }
 
-  return theme('table', array('header' => $header, 'rows' => $rows));
+  return theme('table', $header, $rows);
 }
 
 /**
  * Form builder for the source code bulk parser form.
  */
-function gpui_bulk_parser_form($form, &$form_state) {
+function gpui_bulk_parser_form($form_state) {
   gpui_check_settings($form_state);
 
   // Set default values.
@@ -568,7 +567,7 @@ function gpui_bulk_parser_form_defaults($form_state) {
 /**
  * Form builder for the source code inline parser form.
  */
-function gpui_inline_parser_form($form, &$form_state) {
+function gpui_inline_parser_form($form_state) {
 //   gpui_check_settings();
 
   // Set default values.
