diff --git a/modules/maestro_common/maestro_common.module b/modules/maestro_common/maestro_common.module
index 1d7879b..6ddbebd 100644
--- a/modules/maestro_common/maestro_common.module
+++ b/modules/maestro_common/maestro_common.module
@@ -176,7 +176,9 @@ function maestro_accept_reject($op,&$task,$parms) {
         $maestro = Maestro::createMaestroObject(1);
         $parms['message'] = $maestro->engine()->getProcessVariable('message', $task->_properties->process_id);
       }
-      $retval->html = drupal_render(drupal_get_form('maestro_show_message_accept_reject_form',array('message' => $parms['message'])));
+      $accept_reject_form = drupal_get_form('maestro_show_message_accept_reject_form',array('message' => $parms['message']));
+
+      $retval->html = drupal_render($accept_reject_form);
       $retval->retcode = TRUE;
       $retval->engineop = '';
       break;
