diff --git a/modules/cart/commerce_cart.module b/modules/cart/commerce_cart.module
index 9b0f458..8d2028c 100644
--- a/modules/cart/commerce_cart.module
+++ b/modules/cart/commerce_cart.module
@@ -1706,7 +1706,17 @@ function commerce_cart_add_to_cart_form($form, &$form_state, $line_item, $show_q
               continue;
             }
 
-            if ($product_wrapper->{$key}->raw() != $value) {
+            // For date field support, START HERE.
+            $product_wrapper_key_data = $product_wrapper->{$key}->raw();
+            if (is_array($product_wrapper_key_data)) {
+              foreach ($product_wrapper_key_data AS $key => $value1) {
+                // This is specific to date
+                if ($key == 'value') {
+                  $product_wrapper_key_data = $value1;
+                }
+              }
+            }
+            if ($product_wrapper_key_data != $value) { // By Yogesh, for date field support, END HERE.
               $match = FALSE;
             }
           }
@@ -1721,6 +1731,7 @@ function commerce_cart_add_to_cart_form($form, &$form_state, $line_item, $show_q
             $matching_product_wrapper = entity_metadata_wrapper('commerce_product', $matching_product);
 
             foreach ($attribute_names as $key => $value) {
+              // We dont need patch here, because both side values will be in same format always
               if ($product_wrapper->{$key}->raw() != $matching_product_wrapper->{$key}->raw()) {
                 $match = FALSE;
               }
@@ -1856,7 +1867,16 @@ function commerce_cart_add_to_cart_form($form, &$form_state, $line_item, $show_q
               }
 
               if (isset($form['attributes'][$used_field_name]['#default_value'])) {
-                if ($product_wrapper->{$used_field_name}->raw() != $form['attributes'][$used_field_name]['#default_value']) {
+                // For date field support, START HERE.
+                $product_wrapper_used_data = $product_wrapper->{$used_field_name}->raw();
+                if (is_array($product_wrapper_used_data)) {
+                  foreach ($product_wrapper_used_data AS $key => $value) {
+                    if ($key == 'value') {
+                      $product_wrapper_used_data = $value;
+                    }
+                  }
+                }
+                if ($product_wrapper_used_data != $form['attributes'][$used_field_name]['#default_value']) { // By Yogesh, for date field support, END HERE.
                   continue 2;
                 }
               }
@@ -1870,6 +1890,28 @@ function commerce_cart_add_to_cart_form($form, &$form_state, $line_item, $show_q
             $used_options[$field_name][] = $product_wrapper->{$field_name}->raw();
           }
 
+          // For date field support, START HERE.
+          $default_used_options[$field_name] = $default_product_wrapper->{$field_name}->raw();
+          if (is_array($used_options[$field_name][0])) {
+            $new_used_options = $used_options[$field_name];
+            $used_options[$field_name] = array();
+            foreach ($new_used_options AS $key => $value) {
+              if (is_array($value)) {
+                $used_options[$field_name][] = $value['value'];
+              }
+            }
+
+            // We also need to change other variable which play very important role in form
+            $new_default_used_options = $default_used_options[$field_name];
+            $default_used_options[$field_name] = '';
+            foreach ($new_default_used_options AS $key => $value) {
+              if ($key == 'value') {
+                $default_used_options[$field_name] = $new_default_used_options[$key];
+              }
+            }
+          }
+          // For date field support, END HERE.
+
           // If for some reason no options for this field are used, remove it
           // from the qualifying fields array.
           if (empty($field_has_options[$field_name]) || empty($used_options[$field_name])) {
@@ -1880,7 +1922,7 @@ function commerce_cart_add_to_cart_form($form, &$form_state, $line_item, $show_q
               '#type' => $data['commerce_cart_settings']['attribute_widget'],
               '#title' => commerce_cart_attribute_widget_title($data['instance']),
               '#options' => array_intersect_key($data['options'], drupal_map_assoc($used_options[$field_name])),
-              '#default_value' => $default_product_wrapper->{$field_name}->raw(),
+              '#default_value' => $default_used_options[$field_name], // $default_product_wrapper->{$field_name}->raw(), // By Yogesh, for date field support.
               '#weight' => $data['instance']['widget']['weight'],
               '#ajax' => array(
                 'callback' => 'commerce_cart_add_to_cart_form_attributes_refresh',
@@ -1895,7 +1937,7 @@ function commerce_cart_add_to_cart_form($form, &$form_state, $line_item, $show_q
 
             $form['unchanged_attributes'][$field_name] = array(
               '#type' => 'value',
-              '#value' => $default_product_wrapper->{$field_name}->raw(),
+              '#value' => $default_used_options[$field_name], // $default_product_wrapper->{$field_name}->raw(), // By Yogesh, for date field support.
             );
           }
         }
@@ -1920,7 +1962,16 @@ function commerce_cart_add_to_cart_form($form, &$form_state, $line_item, $show_q
               $match = TRUE;
 
               foreach (element_children($form['attributes']) as $field_name) {
-                if ($product_wrapper->{$field_name}->raw() != $form['attributes'][$field_name]['#default_value']) {
+                // For date field support, START HERE.
+                $product_wrapper_field_data = $product_wrapper->{$field_name}->raw();
+                if (is_array($product_wrapper_field_data)) {
+                  foreach ($product_wrapper_field_data AS $key => $value) {
+                    if ($key == 'value') {
+                      $product_wrapper_field_data = $value;
+                    }
+                  }
+                }
+                if ($product_wrapper_field_data != $form['attributes'][$field_name]['#default_value']) { // By Yogesh, for date field support, END HERE.
                   $match = FALSE;
                 }
               }
