Index: modules/system/system.module
===================================================================
RCS file: /cvs/drupal/drupal/modules/system/system.module,v
retrieving revision 1.977
diff -u -p -r1.977 system.module
--- modules/system/system.module	7 Oct 2010 04:58:45 -0000	1.977
+++ modules/system/system.module	7 Oct 2010 20:42:41 -0000
@@ -2276,6 +2276,9 @@ function _system_rebuild_module_data() {
   // Install profile hooks are always executed last.
   $modules[$profile]->weight = 1000;
 
+  // Allow the basic module information to be altered.
+  drupal_alter('system_modules', $modules);
+
   // Set defaults for module info.
   $defaults = array(
     'dependencies' => array(),
