--- includes/bootstrap.inc  2012-06-23 13:59:49.127470837 +0800
+++ includes/database/oci/tests/bootstrap.inc  2012-06-23 14:24:26.740357276 +0800
@@ -2848,12 +2848,7 @@ function ip_address() {
 function drupal_get_schema($table = NULL, $rebuild = FALSE) {
   static $schema;

-  if ($rebuild || !isset($table)) {
-    $schema = drupal_get_complete_schema($rebuild);
-  }
-  elseif (!isset($schema)) {
-    $schema = new SchemaCache();
-  }
+  $schema = database_test_schema();

   if (!isset($table)) {
     return $schema;
