diff --git advanced_help.module advanced_help.module
index ad55411..0486812 100644
--- advanced_help.module
+++ advanced_help.module
@@ -769,6 +769,8 @@ function _advanced_help_parse_ini() {
           $file = !empty($topic['file']) ? $topic['file'] : $name;
           $cache['topics'][$module][$name] = array(
             'name' => $name,
+            'module' => $module,
+            'ini' => $topic,
             'title' => !empty($translation[$name]['title']) ? $translation[$name]['title'] : $topic['title'],
             'weight' => isset($topic['weight']) ? $topic['weight'] : 0,
             'parent' => isset($topic['parent']) ? $topic['parent'] : 0,
