--- profiles/standard/standard.install.orig	2012-01-02 11:05:35.546510695 -0600
+++ profiles/standard/standard.install	2011-12-30 23:57:51.306513689 -0600
@@ -190,6 +190,12 @@ function standard_install() {
   );
   $query = db_insert('block')->fields(array('module', 'delta', 'theme', 'status', 'weight', 'region', 'pages', 'cache'));
   foreach ($values as $record) {
+    db_delete('block')
+      ->condition('module', $record['module'])
+      ->condition('delta', $record['delta'])
+      ->condition('theme', $record['theme'])
+      ->execute();
+
     $query->values($record);
   }
   $query->execute();
