diff --git a/commerce_stripe.module b/commerce_stripe.module
index dd13bbf..ea04e49 100755
--- a/commerce_stripe.module
+++ b/commerce_stripe.module
@@ -546,6 +546,8 @@ function commerce_stripe_cardonfile_charge($payment_method, $card_data, $order,
     }
   }
   catch (Exception $e) {
+    drupal_set_message(t('We received the following error processing your card. Please enter your information again or try a different card.'), 'error');
+    drupal_set_message(check_plain($e->getMessage()), 'error');
     watchdog('commerce_stripe', 'Following error received when processing card @stripe_error.', array('@stripe_error' => $e->getMessage()), WATCHDOG_NOTICE);
     $transaction->remote_id = $e->getHttpStatus();
     $transaction->payload[REQUEST_TIME] = $e->jsonBody;
