diff --git a/page_rebuild_cache.module b/page_rebuild_cache.module
index 944d71f..014b14a 100644
--- a/page_rebuild_cache.module
+++ b/page_rebuild_cache.module
@@ -26,6 +26,7 @@ function page_rebuild_cache_menu() {
 
   // Admin settings
   $items['admin/page_rebuild_cache_trigger'] = array(
+    'title' => 'Rebuild page cache',
     'page callback' => 'page_rebuild_cache_trigger',
     'access arguments' => array('page rebuild cache'),
   );
