Index: xc_metadata.module
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/xc/xc_metadata/xc_metadata.module,v
retrieving revision 1.22.2.75
diff -c -r1.22.2.75 xc_metadata.module
*** xc_metadata.module	12 Jul 2010 14:30:29 -0000	1.22.2.75
--- xc_metadata.module	2 Sep 2010 12:48:26 -0000
***************
*** 2769,2774 ****
--- 2769,2777 ----
    }
    if (xc_metadata_field_exists($field['name'], $field['namespace'])) {
      $qualified = _array_to_qualified_name($field, 'field');
+     if (is_null($alter) && !isset($object->metadata[$qualified])) {
+       return NULL;
+     }
      if (in_array('all', $delta)) {
        $delta = array_keys($object->metadata[$qualified]);
      }
