diff --git a/includes/bootstrap.inc b/includes/bootstrap.inc
index 744fc8f..fc5c86d 100644
--- a/includes/bootstrap.inc
+++ b/includes/bootstrap.inc
@@ -1044,7 +1044,17 @@ function drupal_page_get_cache($check_only = FALSE) {
   }
 
   if (drupal_page_is_cacheable()) {
-    $cache = cache_get($base_root . request_uri(), 'cache_page');
+    $cid = $base_root . request_uri();
+    // Allow modules to alter the cid. We can't use the regular hook invocation
+    // because this runs in the bootstrap.
+    // @see bootstrap_invoke_all()
+    // @see module_invoke()
+    foreach (module_list() as $module) {
+      if (module_hook($module, 'page_cache_cid_alter')) {
+        call_user_func_array($module . '_page_cache_cid_alter', array(&$cid));
+      }
+    }
+    $cache = cache_get($cid, 'cache_page');
     if ($cache !== FALSE) {
       $cache_hit = TRUE;
     }
diff --git a/includes/common.inc b/includes/common.inc
index 20cc82b..85025b8 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -5215,11 +5215,12 @@ function drupal_page_set_cache() {
     // Check whether the current page might be compressed.
     $page_compressed = variable_get('page_compression', TRUE) && extension_loaded('zlib');
 
+    $body = ob_get_clean();
     $cache = (object) array(
       'cid' => $base_root . request_uri(),
       'data' => array(
         'path' => $_GET['q'],
-        'body' => ob_get_clean(),
+        'body' => $body,
         'title' => drupal_get_title(),
         'headers' => array(),
         // We need to store whether page was compressed or not,
@@ -5230,6 +5231,16 @@ function drupal_page_set_cache() {
       'created' => REQUEST_TIME,
     );
 
+    // Allow modules to alter the cache object.
+    drupal_alter('page_cache_object', $cache);
+
+    // If caching got disabled - fill the output buffer again and return.
+    if (!$cache) {
+      ob_start();
+      echo $body;
+      return FALSE;
+    }
+
     // Restore preferred header names based on the lower-case names returned
     // by drupal_get_http_header().
     $header_names = _drupal_set_preferred_header_name();
-- 
2.2.1

