diff --git a/route_planner.admin.inc b/route_planner.admin.inc
index 3cdd8c3..ac32e77 100644
--- a/route_planner.admin.inc
+++ b/route_planner.admin.inc
@@ -111,5 +111,18 @@ function route_planner_settings_form() {
     '#title'         => t('Enable Overview Map'),
     '#default_value' => variable_get('route_planner_map_overviewmapcontrol', TRUE),
   );
+  $form['map-style'] = array(
+    '#type'  => 'fieldset',
+    '#title' => t('Map Style Settings'),
+    '#description' => t('Optional Style Settings as described at <a target="_blank" href="@google_dev_url">Google DeveloperStyles</a>. You can find a lot of sample styles at <a target="_blank" href="@snazzymaps_url"> snazzymaps.com</a>.', array(
+      '@google_dev_url' => url('https://developers.google.com/maps/documentation/javascript/styling#style_array_example'),
+      '@snazzymaps_url' => url('http://snazzymaps.com/')
+    )),
+  );
+  $form['map-style']['route_planner_map_style'] = array(
+    '#type' => 'textarea',
+    '#title' => 'Style Array',
+    '#default_value' => variable_get('route_planner_map_style'),
+  );
   return system_settings_form($form);
 }
diff --git a/route_planner.install b/route_planner.install
index ddef509..1c84c0e 100644
--- a/route_planner.install
+++ b/route_planner.install
@@ -32,4 +32,5 @@ function route_planner_uninstall() {
   variable_del('route_planner_map_streetviewcontrol');
   variable_del('route_planner_map_overviewmapcontrol');
   variable_del('route_planner_map_defaultui');
+  variable_del('route_planner_map_style');
 }
diff --git a/route_planner.js b/route_planner.js
index be5bc1d..2b308c8 100644
--- a/route_planner.js
+++ b/route_planner.js
@@ -21,6 +21,19 @@
         directionsDisplay = new google.maps.DirectionsRenderer();
         geocoder = new google.maps.Geocoder();
         var latLng;
+        var stylearray = Drupal.settings.routePlanner['style'];
+        var mapstyle = [];
+        if (stylearray) {
+          try {
+            mapstyle = eval(stylearray);
+          } catch (e) {
+            if (e instanceof SyntaxError) {
+              console.log(e.message);
+              // Error on parsing string. Using default.
+              mapstyle = [];
+            }
+          }
+        }
         if (geocoder) {
             var end = Drupal.settings.routePlanner['end'];
             if(document.getElementById("edit-end")){
@@ -44,7 +57,8 @@
                         overviewMapControl: Drupal.settings.routePlanner['overviewmapcontrol'],
                         disableDefaultUI: Drupal.settings.routePlanner['defaultui'],
                         mapTypeId: google.maps.MapTypeId.ROADMAP,
-                        center: location
+                        center: location,
+                        styles: mapstyle
                     }
                     this.map = new google.maps.Map(document.getElementById("map_canvas"), myOptions);
                     var marker = new google.maps.Marker({
diff --git a/route_planner.module b/route_planner.module
index 52d6bb8..4e286d0 100644
--- a/route_planner.module
+++ b/route_planner.module
@@ -76,6 +76,7 @@ function route_planner_block_save($delta = '', $edit = array()) {
       variable_set('route_planner_map_overviewmapcontrol', intval($edit['route_planner_map_overviewmapcontrol']));
       variable_set('route_planner_unitsystem', intval($edit['route_planner_unitsystem']));
       variable_set('route_planner_map_defaultui', intval($edit['route_planner_map_defaultui']));
+      variable_set('route_planner_map_style', $edit['route_planner_map_style']);
       break;
   }
 }
@@ -172,7 +173,8 @@ function route_planner_get_address_form() {
       'overviewmapcontrol'       => variable_get('route_planner_map_overviewmapcontrol', TRUE),
       'unitSystem'               => variable_get('route_planner_unitsystem', TRUE),
       'end'                      => variable_get('route_planner_address', 'Hamburg, Germany'),
-      'defaultui'                => variable_get('route_planner_map_defaultui', TRUE)
+      'defaultui'                => variable_get('route_planner_map_defaultui', TRUE),
+      'style'                    => variable_get('route_planner_map_style', NULL)
     ),
   ), 'setting');

@@ -215,6 +217,7 @@ function route_planner_map_display() {
       'unitSystem'               => variable_get('route_planner_unitsystem', TRUE),
       'defaultui'                => variable_get('route_planner_map_defaultui', TRUE),
       'end'                      => variable_get('route_planner_address', 'Hamburg, Germany'),
+      'style'                    => variable_get('route_planner_map_style', NULL)
     ),
   ), 'setting');

@@ -223,4 +226,3 @@ function route_planner_map_display() {

   return $output;
 }
-
