diff -Nurp quail-lib-0.4.1.orig/quail/common/accessibility_tests.php quail-lib-0.4.1/quail/common/accessibility_tests.php
--- quail-lib-0.4.1.orig/quail/common/accessibility_tests.php	2012-01-19 14:13:20.078372475 -0600
+++ quail-lib-0.4.1/quail/common/accessibility_tests.php	2012-01-19 14:11:44.895039697 -0600
@@ -4547,13 +4547,19 @@ class pNotUsedAsHeader extends quailTest
 				($p->nodeValue == $p->firstChild->nodeValue) &&
 			 	property_exists($p->firstChild, 'tagName') && 
 				in_array($p->firstChild->tagName, $this->head_tags)) {
-					$this->addReport($p);
+					$matches = preg_match("/\.\W*/", $p->nodeValue);
+					if ($matches == 0){
+						$this->addReport($p);
+					}
 					
 			}
 			else {
 				$style = $this->css->getStyle($p);
 				if(isset($style['font-weight']) && $style['font-weight'] == 'bold') {
-					$this->addReport($p);
+					$matches = preg_match("/\.\W*/", $p->nodeValue);
+					if ($matches == 0){
+						$this->addReport($p);
+					}
 				}
 			}
 		}
