diff --git a/fieldset_helper.module b/fieldset_helper.module
index d89a641..49ab8c7 100644
--- a/fieldset_helper.module
+++ b/fieldset_helper.module
@@ -342,6 +342,15 @@ function fieldset_helper_state_manager_get_lookup_id($element_id = NULL) {
 
   $current_path = $_GET['q'];
 
+  // This snippet is inserted from the 6.x-2.x dev version. To work together with
+  // The eXtensible Catalog Drupal Toolkit module, this is needed. We previously
+  // created our own version, but it would be more maintainable to use the module's
+  // owner's implementation. -- xc developers
+  foreach (module_implements('fieldset_helper_path_alter') as $module) {
+    $function = $module .'_fieldset_helper_path_alter';
+    $function($current_path);
+  }
+
   // Load existing lookup ids for the current path from the database.
   if (!isset($lookup_id_table)) {
     // Fetch lookup records for the current path
