diff --git a/misc/autocomplete.js b/misc/autocomplete.js
index 8f7ac60..77588f2 100644
--- a/misc/autocomplete.js
+++ b/misc/autocomplete.js
@@ -29,10 +29,11 @@ Drupal.behaviors.autocomplete = {
  * Prevents the form from submitting if the suggestions popup is open
  * and closes the suggestions popup when doing so.
  */
-Drupal.autocompleteSubmit = function () {
-  return $('#autocomplete').each(function () {
-    this.owner.hidePopup();
-  }).length == 0;
+Drupal.autocompleteSubmit = function () { 
+  $('#autocomplete').each(function () {
+     this.owner.hidePopup(); 
+  }); // }).size() == 0;
+ return 1;
 };
 
 /**
@@ -306,7 +307,9 @@ Drupal.ACDB.prototype.search = function (searchString) {
         }
       },
       error: function (xmlhttp) {
-        alert(Drupal.ajaxError(xmlhttp, db.uri));
+        if (!(xmlhttp.status == 0 && db.uri.match('xc_search/autocomplete'))) {
+         alert(Drupal.ahahError(xmlhttp, db.uri));
+        }
       }
     });
   }, this.delay);
