diff --git a/sites/all/modules/time_tracker/time_tracker.theme.inc b/sites/all/modules/time_tracker/time_tracker.theme.inc
index f2d2b6c..c45cbf8 100644
--- a/sites/all/modules/time_tracker/time_tracker.theme.inc
+++ b/sites/all/modules/time_tracker/time_tracker.theme.inc
@@ -90,8 +90,8 @@ function theme_time_tracker_time_entry_table($variables) {
   $rows[] = array();
   //Setting up the table headers
   $header = array(
-    array('data' => t('User')),
-    array('data' => t('Activity')),
+//    array('data' => t('User')),
+//    array('data' => t('Activity')),
   );
   // If the deductions field is on
   if (variable_get('enable_deductions_field', 0)) {
@@ -135,14 +135,14 @@ function theme_time_tracker_time_entry_table($variables) {
       // Populate the row
       $row = array(//row
         'data' => array( //row data
-          array( //Cell1
-            'data' => user_load($time_entry->uid)->name,
-            'class' => 'time_entry_username',
-          ),
-          array( //Cell2
-            'data' => _time_tracker_get_activity_name($time_entry->activity),
-            'class' => 'time_entry_activity',
-          ),
+//          array( //Cell1
+//            'data' => user_load($time_entry->uid)->name,
+//            'class' => 'time_entry_username',
+//          ),
+//          array( //Cell2
+//            'data' => _time_tracker_get_activity_name($time_entry->activity),
+//            'class' => 'time_entry_activity',
+//          ),
           array( //Cell3
             'data' => _time_tracker_format_hours_to_hours_and_minutes($time_entry->duration),
             'class' => 'time_entry_duration',
