--- search.module-orig	2007-07-26 21:16:48.000000000 +0200
+++ search.module	2007-10-16 18:24:27.274162000 +0200
@@ -1115,7 +1115,27 @@
 
   // Prepare text
   $text = ' '. strip_tags(str_replace(array('<', '>'), array(' <', '> '), $text)) .' ';
+
+	/* expand the list of search keys by reverse lookup for those words 
+	 * in the text, whose stemms are identical to the stemms of the current key words
+	*/
+  foreach (module_implements('stem_list') as $module) {
+	// get the stem for every key as an array of key => stem
+    $key_stem_list = module_invoke($module, 'stem_list', join($keys, ' '));
+	// get also the stem for each word in the text as text_word => text_stem
+    $text_stem_list = module_invoke($module, 'stem_list', $text);
+
+	/* now check which key-stemms can be found in the array of text_stemms.
+	 * Then take the corresponding text_word and 
+	 * put it in list list of key words (search words)
+	 */
+    foreach($key_stem_list as $k => $key_stem) {
+	$text_hits = array_keys($text_stem_list,$key_stem);
+	$keys = array_merge($keys,$text_hits);
+    }
+  }
   array_walk($keys, '_search_excerpt_replace');
+  array_unique($keys);
   $workkeys = $keys;
 
   // Extract a fragment per keyword for at most 4 keywords.
