diff --git a/oa_plupload.features.features_overrides.inc b/oa_plupload.features.features_overrides.inc
index 0b5061a..7c34179 100644
--- a/oa_plupload.features.features_overrides.inc
+++ b/oa_plupload.features.features_overrides.inc
@@ -16,6 +16,8 @@ function oa_plupload_features_override_default_overrides() {
   $overrides["field_instance.node-oa_discussion_post-field_oa_media.widget|type"] = 'media_multiselect';
   $overrides["field_instance.node-oa_event-field_oa_media.widget|module"] = 'media_multiselect';
   $overrides["field_instance.node-oa_event-field_oa_media.widget|type"] = 'media_multiselect';
+  $overrides["field_instance.node-oa_folder-field_oa_media.widget|module"] = 'media_multiselect';
+  $overrides["field_instance.node-oa_folder-field_oa_media.widget|type"] = 'media_multiselect';
   $overrides["field_instance.node-oa_wiki_page-field_oa_media.widget|module"] = 'media_multiselect';
   $overrides["field_instance.node-oa_wiki_page-field_oa_media.widget|type"] = 'media_multiselect';
   $overrides["field_instance.node-oa_worktracker_task-field_oa_media.widget|module"] = 'media_multiselect';
diff --git a/oa_plupload.info b/oa_plupload.info
index 55cd8c3..a76e7ae 100644
--- a/oa_plupload.info
+++ b/oa_plupload.info
@@ -12,6 +12,8 @@ features[features_overrides][] = field_instance.node-oa_discussion_post-field_oa
 features[features_overrides][] = field_instance.node-oa_discussion_post-field_oa_media.widget|type
 features[features_overrides][] = field_instance.node-oa_event-field_oa_media.widget|module
 features[features_overrides][] = field_instance.node-oa_event-field_oa_media.widget|type
+features[features_overrides][] = field_instance.node-oa_folder-field_oa_media.widget|module
+features[features_overrides][] = field_instance.node-oa_folder-field_oa_media.widget|type
 features[features_overrides][] = field_instance.node-oa_wiki_page-field_oa_media.widget|module
 features[features_overrides][] = field_instance.node-oa_wiki_page-field_oa_media.widget|type
 features[features_overrides][] = field_instance.node-oa_worktracker_task-field_oa_media.widget|module
