diff --git a/plugins/views_plugin_query_default.inc b/plugins/views_plugin_query_default.inc
index c6c7649..e09af6a 100644
--- a/plugins/views_plugin_query_default.inc
+++ b/plugins/views_plugin_query_default.inc
@@ -1486,6 +1486,12 @@ class views_plugin_query_default extends views_plugin_query {
           $view->result[] = $item;
         }
 
+        // Let modules modify the view just after executing the query.
+        foreach (module_implements('views_post_execute_query') as $module) {
+          $function = $module . '_views_post_execute_query';
+          $function($view);
+        }
+
         $this->pager->post_execute($view->result);
 
         if ($this->pager->use_count_query() || !empty($view->get_total_rows)) {
