diff --git a/modules/league_table/league_table.module b/modules/league_table/league_table.module
index 6047fdc..0aa5e14 100644
--- a/modules/league_table/league_table.module
+++ b/modules/league_table/league_table.module
@@ -152,19 +152,32 @@ function league_table_league_game_saved_alter($league_game) {
   league_table_tables_save($league_game);
 }
 
-function league_table_league_game_form_save_alter($league_game) {
+function league_table_league_game_result_form_pre_save_alter(&$league_game) {
   league_table_tables_save($league_game);
 }
 
 /**
  * Create/update game statistics
  */
-function league_table_tables_save($league_game) {
+function league_table_tables_save(&$league_game) {
   $settings = variable_get('league_settings');
   $win_p = $settings['points']['win'];
   $tie_p = $settings['points']['tie'];
   $loose_p = $settings['points']['loose'];
-  $statistics = league_table_game_statistics($league_game->league_game_id);
+  $diff = $league_game->score_a - $league_game->score_b;
+  if ($diff > 0) {
+    $league_game->points_a = $win_p;
+    $league_game->points_b = $loose_p;
+  }
+  else if ($diff < 0) {
+    $league_game->points_a = $loose_p;
+    $league_game->points_b = $win_p;
+  }
+  else {
+    $league_game->points_a = $tie_p;
+    $league_game->points_b = $tie_p;
+  }
+  $statistics = league_table_game_statistics($league_game);
   foreach ($statistics as $game_id => $statistic) {
     $tables = league_table_load_by_game($game_id);
     foreach ($statistic as $team_id => $statistic_data) {
@@ -210,15 +223,15 @@ function league_table_tables_save($league_game) {
 /**
  * Return game statistics per team
  */
-function league_table_game_statistics($game_id) {
-  $game = league_game_load($game_id);
-  $played = $game->status == 2 ? 1 : 0;
-  $team_a = $game->team_a;
-  $team_b = $game->team_b;
-  $score_a = $game->override ? $game->override_score_a : $game->score_a;
-  $score_b = $game->override ? $game->override_score_b : $game->score_b;
-  $points_a = $game->override ? $game->points_a : NULL;
-  $points_b = $game->override ? $game->points_b : NULL;
+function league_table_game_statistics($league_game) {
+  $game_id = $league_game->league_game_id;
+  $played = $league_game->status == 2 ? 1 : 0;
+  $team_a = $league_game->team_a;
+  $team_b = $league_game->team_b;
+  $score_a = $league_game->override ? $league_game->override_score_a : $league_game->score_a;
+  $score_b = $league_game->override ? $league_game->override_score_b : $league_game->score_b;
+  $points_a = $league_game->override ? $league_game->override_points_a : $league_game->points_a;
+  $points_b = $league_game->override ? $league_game->override_points_b : $league_game->points_b;
 
   $statistics = array(
     $game_id => array(
