diff --git a/modules/taxonomy/taxonomy.module b/modules/taxonomy/taxonomy.module
index 9be7dfc..b1f61c7 100644
--- a/modules/taxonomy/taxonomy.module
+++ b/modules/taxonomy/taxonomy.module
@@ -202,6 +202,22 @@ function taxonomy_field_extra_fields() {
  *   An array of nids matching the query.
  */
 function taxonomy_select_nodes($tid, $pager = TRUE, $limit = FALSE, $order = array('t.sticky' => 'DESC', 't.created' => 'DESC')) {
+  static $override_function = NULL;
+
+  if (!isset($override_function)) {
+    $override_function = variable_get('taxonomy_select_nodes_override', FALSE);
+    if ($override_function && !function_exists($override_function)) {
+      $override_function = FALSE;
+    }
+  }
+
+  if ($override_function) {
+    $result = $override_function($tid, $pager, $limit, $order);
+    if ($result !== FALSE) {
+      return $result;
+    }
+  }
+
   if (!variable_get('taxonomy_maintain_index_table', TRUE)) {
     return array();
   }
@@ -1103,6 +1119,22 @@ function taxonomy_get_children($tid, $vid = 0) {
  *   depending on the $load_entities parameter.
  */
 function taxonomy_get_tree($vid, $parent = 0, $max_depth = NULL, $load_entities = FALSE) {
+  static $override_function = NULL;
+
+  if (!isset($override_function)) {
+    $override_function = variable_get('taxonomy_get_tree_override', FALSE);
+    if ($override_function && !function_exists($override_function)) {
+      $override_function = FALSE;
+    }
+  }
+
+  if ($override_function) {
+    $result = $override_function($vid, $parent, $max_depth, $load_entities);
+    if ($result !== FALSE) {
+      return $result;
+    }
+  }
+
   $children = &drupal_static(__FUNCTION__, array());
   $parents = &drupal_static(__FUNCTION__ . ':parents', array());
   $terms = &drupal_static(__FUNCTION__ . ':terms', array());
