diff --git a/Image/GraphViz.php b/Image/GraphViz.php
index 84f3485..22e677e 100644
--- a/Image/GraphViz.php
+++ b/Image/GraphViz.php
@@ -354,15 +354,18 @@ class Image_GraphViz
         switch ($command) {
         case 'dot':
         case 'neato':
-            break;
+        case 'osage':
+        case 'circo':
+        case 'twopi':
+        case 'sfdp':
+          break;
         default:
             $command = $this->graph['directed'] ? 'dot' : 'neato';
         }
         $command_orig = $command;
 
-        $command = $this->binPath.(($command == 'dot') ? $this->dotCommand
-                                                       : $this->neatoCommand);
-
+        $command = $this->binPath.$command;
+        
         $command .= ' -T'.escapeshellarg($format)
                     .' -o'.escapeshellarg($outputfile)
                     .' '.escapeshellarg($dotfile)
