diff --git a/commerce_amazon_fulfillment.module b/commerce_amazon_fulfillment.module
index 5449919..c81c9b9 100644
--- a/commerce_amazon_fulfillment.module
+++ b/commerce_amazon_fulfillment.module
@@ -44,6 +44,34 @@ function commerce_amazon_fulfillment_commerce_checkout_complete($order) {
     // Load the line item.
     $line_item = commerce_line_item_load($line_item_id);
 
+    // Get type of shipping provided.
+    if ($line_item->type == 'shipping') {
+      switch ($line_item->data['shipping_service']['name']) {
+        case 'usa_standard':
+          $shipping_speed = 'Standard';
+          break;
+
+        case 'usa_priority':
+          $shipping_speed = 'Priority';
+          break;
+
+        case 'usa_expedited':
+          $shipping_speed = 'Expedited';
+          break;
+
+        case 'outside_of_usa_standard':
+          $shipping_speed = 'Standard';
+          break;
+
+        case 'outside_of_usa_expedited':
+          $shipping_speed = 'Expedited';
+          break;
+
+        default:
+          $shipping_speed = variable_get('amazon_fulfillment_ShippingSpeedCategory', 'Standard');
+          break;
+      }
+    }
     // Check if a line item is a product.
     if ($line_item->type == 'product') {
       // Load relevant product.
@@ -79,7 +107,7 @@ function commerce_amazon_fulfillment_commerce_checkout_complete($order) {
 
   // Create amazon fulfillment order.
   commerce_amazon_fulfillment_create_order($order->order_number,
-    $amazon_items_to_ship, $amazon_address_to_ship_to);
+    $amazon_items_to_ship, $amazon_address_to_ship_to, $shipping_speed);
 }
 
 /**
diff --git a/includes/commerce_amazon_fulfillment.utility.inc b/includes/commerce_amazon_fulfillment.utility.inc
index efa1be3..d582500 100644
--- a/includes/commerce_amazon_fulfillment.utility.inc
+++ b/includes/commerce_amazon_fulfillment.utility.inc
@@ -121,10 +121,13 @@ function commerce_amazon_fulfillment_create_field_instance($bundle) {
  *
  * @param string $address
  *   A shipping address to send the package(s) to.
+ *
+ * @param string $shipping_speed
+ *   The shipping speed to use with amazon fulfillment.
  */
-function commerce_amazon_fulfillment_create_order($order_id, $items, $address) {
+function commerce_amazon_fulfillment_create_order($order_id, $items, $address, $shipping_speed) {
   // Get data for creating an amazon fulfillment request.
-  $data = commerce_amazon_fulfillment_create_order_data($order_id, $items, $address);
+  $data = commerce_amazon_fulfillment_create_order_data($order_id, $items, $address, $shipping_speed);
   $request_url = variable_get('amazon_fulfillment_marketplace_endpoint');
 
   // Send request to amazon for fulfillment.
@@ -197,7 +200,7 @@ function commerce_amazon_fulfillment_create_order_data($order_id, $items, $addre
     'SellerFulfillmentOrderId' => $order_id,
     'DisplayableOrderId' => variable_get('amazon_fulfillment_DisplayableOrderIdPrefix', '') . $order_id,
     'DisplayableOrderComment' => variable_get('amazon_fulfillment_displayable_comment'),
-    'ShippingSpeedCategory' => variable_get('amazon_fulfillment_ShippingSpeedCategory', 'Standard'),
+    'ShippingSpeedCategory' => $shipping_speed,
     'DestinationAddress.Name' => commerce_amazon_fulfillment_strip_special_characters($address['name_line']),
     'DestinationAddress.Line1' => commerce_amazon_fulfillment_strip_special_characters($address['thoroughfare']),
     'DestinationAddress.CountryCode' => $address['country'],
