diff --git a/includes/common.inc b/includes/common.inc
index 27fa190..36a59a3 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -2013,6 +2013,9 @@ function format_date($timestamp, $type = 'medium', $format = '', $timezone = NUL
   // Set the time zone for the DateTime object.
   date_timezone_set($date_time, $timezones[$timezone]);
 
+  // Preserve format for alter_hook.
+  $original_format = $format;
+
   // Encode markers that should be translated. 'A' becomes '\xEF\AA\xFF'.
   // xEF and xFF are invalid UTF-8 sequences, and we assume they are not in the
   // input string.
@@ -2027,7 +2030,19 @@ function format_date($timestamp, $type = 'medium', $format = '', $timezone = NUL
   _format_date_callback(NULL, $langcode);
 
   // Translate the marked sequences.
-  return preg_replace_callback('/\xEF([AaeDlMTF]?)(.*?)\xFF/', '_format_date_callback', $format);
+  $date = preg_replace_callback('/\xEF([AaeDlMTF]?)(.*?)\xFF/', '_format_date_callback', $format);
+
+  // Give other modules opportunity to change date format
+  $context = array(
+    'timestamp' => $timestamp, 
+    'type' => $type, 
+    'format' => $original_format, 
+    'timezone' => $timezone, 
+    'langcode' => $langcode
+  );
+  drupal_alter('datex_format', $date, $context);
+
+  return $date;
 }
 
 /**
